### PR TITLE
Make knative-build controller set BuildID and URL

### PR DESCRIPTION
**Pull request by fejta**
Also:
* Move JobURL to pjutil so other controllers can reuse it
* Create and auto-set area/prow/knative-build labels

/assign @krzyzacy

### CODE REVIEW

File: prow/cmd/build/controller.go [hunk 1]
```diff
@@ -56,11 +57,13 @@ const (
...
 )
 
+type getConfig func() *config.Config
+
 type controller struct {
-	pjNamespace string
-	pjc         prowjobset.Interface
-	builds      map[string]buildConfig
-	totURL      string
+	config getConfig
```
Comment (krzyzacy) [inline prow/cmd/build/controller.go]: What's the advantage compare with taking in the config agent? Do we want to unify the behavior across all our controllers?

**Comment by stevekuznetsov** [comment]
/lgtm

Would prefer to pass around the config agent or make the change everywhere

**Comment by fejta** [comment]
ref https://github.com/kubernetes/test-infra/issues/10943

**Pull request merged**